test/common/init/target_impl_test.cc

```diff
@@ -60,6 +62,110 @@ TEST(InitTargetImplTest, ReadyWhenWatcherUnavailable) {
   EXPECT_FALSE(target.ready());
 }
 
+TEST(SharedTargetImplTest, OnSyncManagers) {
```

Comment:
Can you use more descriptive names for these tests, e.g. something like `InitializeTwiceBeforeReady` for this one?

Reply:
Will do